### PR TITLE
feat: adds layout plugins

### PR DESCRIPTION
**Pull request by supproduction**
Adds layout plugins

closes [HRZ-89952](https://spryker.atlassian.net/browse/HRZ-89952) , [HRZ-90192](https://spryker.atlassian.net/browse/HRZ-90192)

[HRZ-89952]: https://spryker.atlassian.net/browse/HRZ-89952?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[HRZ-90192]: https://spryker.atlassian.net/browse/HRZ-90192?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/template/presets/storefront/experience/pages/category-page.ts [hunk 1]
```diff
@@ -40,9 +40,18 @@ export const categoryPage: ExperienceComponent = {
...
+            layout:
+              featureVersion >= '1.2'
+                ? {
+                    type: 'grid',
+                    divider: true,
+                    sticky: true,
+                  }
+                : 'grid',
             gap: '1px',
+            // @deprecated since 1.2 will be removed.
```
Comment (dunqan) [inline libs/template/presets/storefront/experience/pages/category-page.ts]: do we need "deprecated values" in sample data? 
can't we just put them behing featuteVersion flag?

File: libs/platform/experience/src/mixins/layout.mixin.ts [hunk 2]
```diff
@@ -46,48 +81,132 @@ export const LayoutMixin = <T extends Type<LitElement & LayoutAttributes>>(
...
+        const attrName = record.attributeName!;
+        (this as Record<string, unknown>)[attrName] =
+          this.getAttribute(attrName);
+      });
+
+      this.observer.disconnect();
+      this.observe();
+    });
+
+    protected observe(layoutSpecificAttrs = []): void {
```
Comment (dunqan) [inline libs/platform/experience/src/mixins/layout.mixin.ts:129]: as mixins are adding properties to component that can conflict with other mixing, can you consider hiding internal implementation ones? 
we are testing this kind of approach: https://github.com/spryker/oryx/blob/feature/marketplace/libs/domain/product/src/merchant/mixins/merchant.mixin.ts#L37

File: libs/platform/experience/src/mixins/layout.mixin.ts [hunk 3]
```diff
@@ -46,48 +81,132 @@ export const LayoutMixin = <T extends Type<LitElement & LayoutAttributes>>(
...
-      type: Boolean,
-      reflect: true,
-      attribute: 'layout-vertical',
-    })
-    vertical?: boolean;
-
-    @signalProperty({ type: Object, reflect: true }) xs?: LayoutProperties;
+    constructor() {
+      super();
+      this.observe();
```
Comment (dunqan) [inline libs/platform/experience/src/mixins/layout.mixin.ts]: should we do it in the constructor?
(doesn't make sense on the SSR also)

Comment (supproduction) [reply libs/platform/experience/src/mixins/layout.mixin.ts]: Yeah, you are right. It's leftover, we have it in connectedCallback

File: libs/template/presets/storefront/experience/footer.ts [hunk 4]
```diff
@@ -44,10 +44,10 @@ const legalLinks: ExperienceComponent = {
                 divider: true,
               }
             : 'flex',
-        // @deprecated since 1.2 will be removed.
         divider: true,
```
Comment (dunqan) [inline libs/template/presets/storefront/experience/footer.ts:47]: this one should be also dropped?

**Pull request merged**